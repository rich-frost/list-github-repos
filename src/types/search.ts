export interface SearchNode {
  id: string;
  name: string;
  url: string;
  stargazerCount: number;
  forkCount: number;
}

export interface SearchEdge {
  node: SearchNode;
}

export interface SearchResults {
  repositoryCount: number;
  edges: SearchEdge[];
}
