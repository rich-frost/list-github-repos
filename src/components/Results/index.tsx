import { SearchEdge } from '../../types/search';
import Repositories from '../Repositories';
import ResultsInfo from '../ResultsInfo';

type ResultsProps = {
  count: number;
  topic: string;
  data: SearchEdge[];
};

export default function Results({ count, topic, data }: ResultsProps) {
  return (
    <>
      <ResultsInfo {...{ count, topic }} />
      <Repositories {...{ data }} />
    </>
  );
}
