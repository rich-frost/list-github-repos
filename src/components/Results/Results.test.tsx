import renderer from 'react-test-renderer';
import { SearchEdge } from '../../types/search';
import Results from './index';

describe('Results component', () => {
  it('renders the component', () => {
    const data: SearchEdge[] = [
      {
        node: {
          id: '123',
          url: 'googl.com',
          name: 'React repo',
          stargazerCount: 201,
          forkCount: 214,
        },
      },
    ];
    const component = renderer.create(
      <Results count={123} topic="React" data={data} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
