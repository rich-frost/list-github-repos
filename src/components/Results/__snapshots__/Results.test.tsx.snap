// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component renders the component 1`] = `
[
  <div
    className="css-1w3riw2"
  >
    <p
      className="chakra-text css-0"
    >
      <span
        className="chakra-badge css-ifsvay"
      >
        123
      </span>
       
      respositories found for topic 
      <b
        className="chakra-text css-0"
      >
        React
      </b>
    </p>
  </div>,
  <div
    className="chakra-table__container css-zipzvv"
  >
    <table
      className="chakra-table css-0"
    >
      <thead
        className="css-0"
      >
        <tr
          className="css-0"
        >
          <th
            className="css-0"
          >
            Name
          </th>
          <th
            className="css-0"
          >
            <svg
              className="chakra-icon css-13otjrl"
              fill="currentColor"
              focusable={false}
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 14 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74L14 6z"
                fillRule="evenodd"
              />
            </svg>
             Stars
          </th>
          <th
            className="css-0"
          >
            <svg
              className="chakra-icon css-13otjrl"
              fill="currentColor"
              focusable={false}
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 10 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 1a1.993 1.993 0 0 0-1 3.72V6L5 8 3 6V4.72A1.993 1.993 0 0 0 2 1a1.993 1.993 0 0 0-1 3.72V6.5l3 3v1.78A1.993 1.993 0 0 0 5 15a1.993 1.993 0 0 0 1-3.72V9.5l3-3V4.72A1.993 1.993 0 0 0 8 1zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3 10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3-10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"
                fillRule="evenodd"
              />
            </svg>
             Forks
          </th>
        </tr>
      </thead>
      <tbody
        className="css-0"
      >
        <tr
          className="css-0"
        >
          <td
            className="css-0"
          >
            <a
              className="chakra-link css-0"
              href="googl.com"
              rel="noopener"
              target="_blank"
            >
              React repo
               
              <svg
                className="chakra-icon css-16o4ymj"
                fill="currentColor"
                focusable={false}
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m13 3 3.293 3.293-7 7 1.414 1.414 7-7L21 11V3z"
                />
                <path
                  d="M19 19H5V5h7l-2-2H5c-1.103 0-2 .897-2 2v14c0 1.103.897 2 2 2h14c1.103 0 2-.897 2-2v-5l-2-2v7z"
                />
              </svg>
            </a>
          </td>
          <td
            className="css-0"
          >
            <svg
              className="chakra-icon css-13otjrl"
              fill="currentColor"
              focusable={false}
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 14 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74L14 6z"
                fillRule="evenodd"
              />
            </svg>
             
            201
          </td>
          <td
            className="css-0"
          >
            <svg
              className="chakra-icon css-13otjrl"
              fill="currentColor"
              focusable={false}
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 10 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 1a1.993 1.993 0 0 0-1 3.72V6L5 8 3 6V4.72A1.993 1.993 0 0 0 2 1a1.993 1.993 0 0 0-1 3.72V6.5l3 3v1.78A1.993 1.993 0 0 0 5 15a1.993 1.993 0 0 0 1-3.72V9.5l3-3V4.72A1.993 1.993 0 0 0 8 1zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3 10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3-10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"
                fillRule="evenodd"
              />
            </svg>
             
            214
          </td>
        </tr>
      </tbody>
      <tfoot
        className="css-0"
      >
        <tr
          className="css-0"
        >
          <td
            className="css-0"
            colSpan={3}
          >
            <div
              className="css-1bvc4cc"
            >
              <i
                className="chakra-text css-0"
              >
                Showing top 10 GitHub repos for topic, sorted by best match
              </i>
            </div>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>,
]
`;
